core/src/main/java/bisq/core/trade/TradeManager.java

```diff
@@ -681,27 +702,46 @@ public boolean isBuyer(Offer offer) {
         return available.filter(addressEntry -> btcWalletService.getBalanceForAddress(addressEntry.getAddress()).isPositive());
     }
 
-    public Stream<Trade> getLockedTradesStream() {
+    public Stream<Trade> getTradesStreamWithFundsLockedIn() {
         return getTradableList().stream()
                 .filter(Trade::isFundsLockedIn);
     }
 
-    public Set<String> getSetOfAllTradeIds() {
-        Set<String> tradesIdSet = getLockedTradesStream()
+    public Set<String> getSetOfFailedOrClosedTradeIdsFromLockedInFunds() throws TradeTxException {
+        AtomicReference<TradeTxException> tradeTxException = new AtomicReference<>();
+        Set<String> tradesIdSet = getTradesStreamWithFundsLockedIn()
                 .filter(Trade::hasFailed)
                 .map(Trade::getId)
                 .collect(Collectors.toSet());
-        tradesIdSet.addAll(failedTradesManager.getLockedTradesStream()
-                .map(Trade::getId)
+        tradesIdSet.addAll(failedTradesManager.getTradesStreamWithFundsLockedIn()
+                .filter(trade -> trade.getDepositTx() != null)
+                .map(trade -> {
+                    log.warn("We found a failed trade with locked up funds. " +
+                            "That should never happen. trade ID=" + trade.getId());
+                    return trade.getId();
+                })
                 .collect(Collectors.toSet()));
-        tradesIdSet.addAll(closedTradableManager.getLockedTradesStream()
-                .map(e -> {
-                    log.warn("We found a closed trade with locked up funds. " +
-                            "That should never happen. trade ID=" + e.getId());
-                    return e.getId();
+        tradesIdSet.addAll(closedTradableManager.getTradesStreamWithFundsLockedIn()
+                .map(trade -> {
+                    Transaction depositTx = trade.getDepositTx();
+                    if (depositTx != null) {
+                        TransactionConfidence confidence = btcWalletService.getConfidenceForTxId(depositTx.getHashAsString());
+                        if (confidence != null && confidence.getConfidenceType() != TransactionConfidence.ConfidenceType.BUILDING) {
+                            tradeTxException.set(new TradeTxException(Res.get("error.closedTradeWithUnconfirmedDepositTx", trade.getShortId())));
+                        } else {
+                            log.warn("We found a closed trade with locked up funds. " +
+                                    "That should never happen. trade ID=" + trade.getId());
+                        }
+                    } else {
+                        tradeTxException.set(new TradeTxException(Res.get("error.closedTradeWithNoDepositTx", trade.getShortId())));
```

Comment:
This means there are no funds locked in, should it still be part of the tradesIdSet representing locked in funds?

Reply:
Hm... Not sure. But after the pending trade has been moved to failed trade I think the tradesIdSet is correct again. All those cases are very hard to reproduce/test...

Reply:
Looks like it will work due to how the result of `getSetOfFailedOrClosedTradeIdsFromLockedInFunds` is used, but perhaps the function name is a bit misguiding since there is no locked in fund. Could return `null` there and filter out nulls before collect().